avaframe/ana4Stats/probAna.py

```diff
@@ -40,31 +43,105 @@ def createComModConfig(cfgProb, avaDir, modName, cfgFileMod=''):
+
+    return cfgFiles, outDir
+
+
+def cfgFilesGlobalApproach(cfgProb, modName, outDir, cfgFileMod):
+    """ create configuration files with all parameter variations - drawn from full sample
+        for performing sims with modName comModule
+
+        Parameters
+        -----------
+        cfgProb: configParser object
+            configuration settings
+        avaDir: pathlib path
+            path to avalanche directory
+        modName: module
+            computational module
+        cfgFileMod: str
+            path to cfgFile for computational module - optional
+
+        Returns
+        -------
+        cfgFiles: list
+            list of paths to newly generated configuration files for com module inlcuding parameter
+            variations
+    """
+
+    # create sample of all parameter variations
+    paramValuesD = createSampleWithPercentVariation(cfgProb, modName, fileOverride=cfgFileMod)
+
+    # create plot of parameter sample if variation of two parameters
+    sP.plotSample(paramValuesD, outDir)
+
+    # write cfg files one for each parameter set drawn from full sample
+    cfgFiles = createCfgFiles(paramValuesD, modName, cfgProb, cfgPath=outDir,
+                              fileOverride=cfgFileMod)
+
+    return cfgFiles
+
+
+def cfgFilesLocalApproach(cfgProb, modName, outDir, cfgFileMod):
+    """ create configuration file for performing sims with modName com module
+
+        Parameters
+        -----------
+        cfgProb: configParser object
+            configuration settings
+        avaDir: pathlib path
+            path to avalanche directory
+        modName: module
+            computational module
+        cfgFileMod: str
+            path to cfgFile for computational module name - optional
+
+        Returns
+        -------
+        cfgFiles: dict
+            dictionary of paths to newly generated configuration files for com module for all parameters
+
+    """
+
     # loop over all parameters for performing parameter variation
     variationsDict = makeDictFromVars(cfgProb['PROBRUN'])
-    cfgFiles = {}
+    cfgFiles = []
     for varName in variationsDict:
         # define configuration files
         # get filename of module
         modNameString = str(pathlib.Path(modName.__file__).stem)
         cfgFile = outDir / ('probRun%sCfg%s.ini' % (modNameString, varName))
 
         # use cfgFile, local com module settings or default settings if local not available
-        modCfg = cfgUtils.getModuleConfig(modName, fileOverride=cfgFileMod)
+        if cfgFileMod != '' and cfgProb['PROBRUN'].getboolean('defaultComModuleCfg'):
+            message = 'fileOverride provided AND defaultComModuleCfg set to True, only one is allowed'
+            log.error(message)
+            raise AssertionError(message)
+        else:
+            modCfg = cfgUtils.getModuleConfig(modName, fileOverride=cfgFileMod, onlyDefault=cfgProb['PROBRUN'].getboolean('defaultComModuleCfg'))
```

Comment:
This code is basically repeated 3 times; I suggest we either put it in a function or try to address it before the cfgFilesXXXApproach functions are called...

Reply:
moved to separate function